### PR TITLE
feat: add initial rendering implementation

### PR DESCRIPTION
**Pull request by DmitryDodzin**

### CODE REVIEW

File: src/toyer-video.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        width: width ?? context.canvas.width,
+        height: height ?? context.canvas.height,
+      }),
+    [context, currentIndex, width, height, top, left],
+  ); // Its intentinal to keep out playing from deps here
+
+  return null;
+};
+
+ToyerVideo.displayName = 'ToyerVideo';
```
Comment (roikoren755) [inline src/toyer-video.tsx:45]: why?

File: src/toyer.tsx [hunk 2]
```diff
@@ -1,5 +1,65 @@
...
+
+  return (
+    <div style={{ height, width }}>
+      <canvas ref={canvasRef} height={height} width={width} />
+      <ToyerContext.Provider value={contextRef.current}>{children}</ToyerContext.Provider>
+    </div>
+  );
+};
+
+Toyer.displayName = 'Toyer';
```
Comment (roikoren755) [inline src/toyer.tsx:74]: delete

File: package.json [hunk 3]
```diff
@@ -41,6 +41,7 @@
     "@testing-library/react": "12.0.0",
     "@types/jest": "26.0.24",
     "@types/react": "17.0.14",
+    "canvas": "2.8.0",
```
Comment (roikoren755) [inline package.json:44]: what?

Comment (roikoren755) [reply package.json:44]: this isn't what you think it is, probably, and isn't actually used anywhere

File: src/toyer.helpers.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
+import type { IToyerContext, IVideoItem, ValueType } from './interfaces';
+
+export const registerVideo = (context: IToyerContext, video: IVideoItem): (() => void) => {
+  if (video.element.readyState === 4) {
```
Comment (joni7777) [inline src/toyer.helpers.ts:4]: whats 4?

File: .changeset/quick-masks-clean.md [hunk 5]
```diff
@@ -0,0 +1,14 @@
+---
+"react-toyer": patch
+---
+
+#### Api
+
+feat: create initial api for rendering several vidos on one canvas
+
+feat: FramerMotion's motionValue support for canvas internals (not canvas itself)
```
Comment (roikoren755) [inline .changeset/quick-masks-clean.md:9]: FramerMotion? what?

File: package.json (hunk 3 above)
Comment (DmitryDodzin) [reply package.json:44]: Test's won't render without this, it's a headless canvas implementation

File: src/toyer.helpers.ts (hunk 4 above)
Comment (DmitryDodzin) [reply src/toyer.helpers.ts:4]: Ready

File: .changeset/quick-masks-clean.md (hunk 5 above)
Comment (DmitryDodzin) [reply .changeset/quick-masks-clean.md:9]: yes

File: src/toyer-video.tsx (hunk 1 above)
Comment (DmitryDodzin) [reply src/toyer-video.tsx:45]: There is no autogenerated ones in the js files

File: src/toyer.helpers.ts (hunk 4 above)
Comment (roikoren755) [reply src/toyer.helpers.ts:4]: create a const enum somewhere with the different states, or export a named const for each state please

File: src/toyer-video.tsx (hunk 1 above)
Comment (roikoren755) [reply src/toyer-video.tsx:45]: doesn't need to be one

File: package.json (hunk 3 above)
Comment (roikoren755) [reply package.json:44]: what

File: .changeset/quick-masks-clean.md (hunk 5 above)
Comment (roikoren755) [reply .changeset/quick-masks-clean.md:9]: what

File: src/toyer-video.tsx (hunk 1 above)
Comment (DmitryDodzin) [reply src/toyer-video.tsx:45]: for us yes not for everyone

File: .changeset/quick-masks-clean.md (hunk 5 above)
Comment (DmitryDodzin) [reply .changeset/quick-masks-clean.md:9]: yes

File: package.json (hunk 3 above)
Comment (DmitryDodzin) [reply package.json:44]: try and run test without it

File: src/toyer.helpers.ts [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+    context.videos[video.index] = video;
```
Comment (joni7777) [inline src/toyer.helpers.ts:5]:
```suggestion

CONST VIDEO_READY = 4;

export const registerVideo = (context: IToyerContext, video: IVideoItem): (() => void) => {
  if (video.element.readyState === VIDEO_READY) {
    context.videos[video.index] = video;
```

**Review by joni7777 — APPROVED** [review]

**Pull request merged**